pytorch_lightning/strategies/horovod.py

```diff
@@ -112,7 +118,13 @@ def _unpack_lightning_optimizer(opt):
         for optimizer in optimizers:
             hvd.broadcast_optimizer_state(optimizer, root_rank=0)
 
-        self.optimizers = self._wrap_optimizers(optimizers)
+        accumulation_scheduler = trainer.accumulation_scheduler
+        if accumulation_scheduler.epochs != [0]:
+            raise MisconfigurationException(
```

Comment:
Please add a test for this

Reply:
Thanks, @ananthsub for the suggestion. I've added the tests, but instead of using `_run_horovod`, I went for `trainer.fit(model)` since `_run_horovod` dumps the `dict` into a string object (and does not return the keys as integers, so the test will fail for this reason). I hope the newly added tests look good now.